### DISCUSSION TITLE
Upload Image using python

### DISCUSSION THREAD

**Discussion by yarinc99** [etsy/open-api#923] [Q&A] [answered]
hello, i have been trying to upload an image to a draft listing using python and i keep getting the same error message,
"Either a valid image file or a listing_image_id must be provided."
I have tried to upload the image using the "with open('.png/.jpg')" and also base64 encoding it, and it didnt work.
If someone succeeded in doing it and can please share the code or have a quick fix !!!
(Even Javascript code that works OR a successfull HTTP request will be helpful !!!!)

Thank you for the helpers !!!!

**Comment by itsdarrylnorris** [comment]
Hey @yarinc99 , Any luck with this?

**Reply by yarinc99** [reply]
Yes i got it to work if you need any help contact me

**Reply by rubberfashion** [reply]
Hi, can you provide the code?

I alway got Error "Either a valid image file or a listing_image_id must be provided"

**Reply by yarinc99** [reply]
If someone has any problem you can private message me your discord and i will help you with the code

**Reply by yarinc99** [reply, marked as answer]
import requests

auth_headers = {'Authorization':f"Bearer {access_token}",'x-api-key':API_KEY}
with open('image.jpg', 'rb') as f:
    image_data = f.read()       
files = {'name':'image.jpg','image': ('image.jpeg', image_data, 'image/jpg')}    
r3 = requests.post(f'https://openapi.etsy.com/v3/application/shops/{shop_id}/listings/{listing_id}/images',files=files,headers=auth_headers)

**Reply by sebing** [reply]
The solution is not working for me ;( 

```
files = {'name': 'image.jpeg', 'image': ('image.jpeg', binary, 'image/jpeg')}
requests.post(url, headers=headers_file,  files=files).json()
{'error': 'Either a valid listing_file_id or file must be provided.'}
```


any trick with the 
```
'Content-Type': 'multipart/form-data'
```
@yarinc99 how can i contact you :)

**Reply by sebing** [reply]
@yarinc99  I have send you a friend requst in discord. Please accept :)

**Reply by sebing** [reply]
@yarinc99 thanks a lot. 
The issue with my code was I had 
```
  'Content-Type': 'multipart/form-data',
  'Accept': 'application/json',
```

all we need to pass is the `x-api-key` and `Authorization`
